dashboard/app/models/circuit_playground_discount_application.rb

```diff
@@ -54,11 +53,11 @@ def self.find_by_studio_person_id(studio_person_id)
   end
 
   # @return {boolean} true if user is an eligible facilitator or attended relevant workshop
-  def self.user_pd_eligible?(user)
-    user_pd_eligible_as_teacher?(user) || user_pd_eligible_as_facilitator?(user)
+  def self.user_pd_eligible?(user, workshop_subjects=nil)
```

Comment:
Why did you choose a nil default for this parameter? That passes through to the query below and would match workshops with a `NULL` subject, which doesn't make sense to me as a default.  If it's possible to simply remove the default and require this parameter, I'd prefer that.

Reply:
I made it nil by default so you could still pass in a facilitator's user ID and get a returned value of true, something like [this](https://github.com/code-dot-org/code-dot-org/pull/32179/files/3a8e9e652e6e95163ad1ddfb18f52c93c7f0fffc#diff-f1db5c0aad5f7af468b0273d4b90ebcbL143).

I didn't consider the impact of matching a workshop with a `null` subject though.

Reply:
Did a quick scan and it looks like we don't ever call this method without a second argument in production code, only in test code.  Let's remove the default value and any tests covering that behavior.

Reply:
Sounds good -- for the record, I think the link I put here wasn't to the right test, more like the one titled "studio_person_pd_eligible? returns true if studio_person_id associated User is approved facilitator" is relevant.

Reply:
Updated -- I kept most tests for facilitators in place, just added the 5-day workshop as the `workshop_subjects` argument. I think this tests what we'd see in the wild (facilitators going through the normal flow with no enrollments associated with their accounts, but with their user ID in the list of facilitators).

Reply:
:+1: Great!